todo_app/data/session_items.py

```diff
@@ -67,3 +67,22 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+def remove_item(id):
+    """
+    Deletes item from collection with the specified ID.
+
+    Args:
+        id: item ID.
+
+    Returns:
+        item: The saved item, or None if no items match the specified ID.
+    """
+    item = get_item(id)
+    existing_items = get_items()
+    if item != None:
+        # item with matching ID found and deleted
+        existing_items.remove(item)
+        session['items'] = existing_items
+
+    return item
```

Comment:
Not so much a suggestion as showing an alternative for the sake of it, but worth comparing this to it's list-comprehension equivalent based on the save_item method:
```python
    existing_items = get_items()
    print(existing_items)
    updated_items = [item for item in existing_items if item['id'] != int(id) ]
    
    session['items'] = updated_items
```
Note this does either force an otherwise-unnecessary call to get_item, or to change the signature of the function (i.e. to change what it returns)